Qt Settings: Transfer OSD settings to a new OSD pane.

TryTwo opened the pull request:
Collects many scattered OSD options into one place. I also did a little more than usual to align the group box columns so the QSpinBoxes look better.

<img width="400" alt="image" src="https://github.com/user-attachments/assets/11676abb-e105-47a6-9b8d-ac1f1c4de86c" />

Looking for feedback.  There are a few OSD options left in Graphics -> Advanced, such as Show Projection Stats.  Not sure if those should stay in the debugging group box in Advanced or be moved. I'm thinking they should stay.

Graphics -> General -> Other only has two items in it now. Maybe those could just be moved to the Box above?

There could be multiple font size options for the various groups. Font size also scales everything up, so not sure if it should be called something else.

The Other Statistics (Movie window) feels like a collection of random stuff. Maybe that could get improved later.

mbc07 commented:
For a first iteration, it's already looking very promising. A few points, though:

> I also did a little more than usual to align the group box columns so the QSpinBoxes look better.

AFAICT this seems to lock the width of the columns to the minimum required to hold the longest string of said column. Initially it looks good, but the side effect is that the groups aren't always aligned (e.g. second column of Netplay doesn't align with the other groups, at least with Dolphin set to Brazilian Portuguese) and that resizing the window becomes way less responsible (e.g. the widgets aren't adjusted anymore to fill the entire available space). IMHO this looks worse than the default behavior of dynamically filling all available space, and makes it out of place in comparison to the other two column layouts we have in that same window (e.g. Graphics, Wii), so I'd revert that.

> There could be multiple font size options for the various groups. Font size also scales everything up, so not sure if it should be called something else.

Do you intend to expose that in the GUI? It sounds a bit cumbersome but perhaps have the other font size options as a multiplier of the main font size? Not sure if this approach makes sense, but I feel it would be clearer to the user the sizes are related to each other by establishing a relation between the font sizes for those different groups...

> Graphics -> General -> Other only has two items in it now. Maybe those could just be moved to the Box above?

I'm neutral to this change. They could be kept as is (perhaps renaming "Other" to something else) or merged with the existing "Basic" group, both are perfectly fine IMO.

Regarding the "Other Statistics" group: should probably be the last one, below "Netplay". Also, the tooltips in that group are empty, I think we should add proper text to them, to keep consistent with the other controls in that pane.

TryTwo commented:
> > I also did a little more than usual to align the group box columns so the QSpinBoxes look better.
> 
>Initially it looks good, but the side effect is that the groups aren't always aligned (e.g. second column of Netplay doesn't align with the other groups, at least with Dolphin set to Brazilian Portuguese) and that resizing the window becomes way less responsible (e.g. the widgets aren't adjusted anymore to fill the entire available space). 

Thanks, I've reverted it to normal.

> > There could be multiple font size options for the various groups. Font size also scales everything up, so not sure if it should be called something else.
> 
> Do you intend to expose that in the GUI? It sounds a bit cumbersome but perhaps have the other font size options as a multiplier of the main font size? Not sure if this approach makes sense, but I feel it would be clearer to the user the sizes are related to each other by establishing a relation between the font sizes for those different groups...

I'm not sure if I was unclear. I meant any font size change scales the U that uses that font size.   Global font size wouldn't need to affect an individually set component's font size.  Thinking about how to code it so that you can just use a global size and not mess with individual sizes is a bit annoying. Though making the sizes multiply each other would make it simple...

I will only include the one global size in this PR.

> Regarding the "Other Statistics" group: should probably be the last one, below "Netplay". Also, the tooltips in that group are empty, I think we should add proper text to them, to keep consistent with the other controls in that pane.

I fixed the tooltips.  I renamed Other Statistics to Movie Window, which I guess does make more sense. Should it still go below Netplay?

TryTwo commented:
Updated the first post. It has new screenshots. I could probably remove Show from each movie window item, except the top one.

TryTwo commented:
> Still giving this thought. In general I like the idea but for very specific things (like netplay and maybe movie?) it might actually be more confusing to split out the setting locations (not sure where those settings are now). So I am a bit torn.

Netplay was in graphics -> general. I actually had trouble finding it to test netplay chat, which was named 'netplay messages'.

> I'll have to go look at what other settings aren't here. It might help for you to explicitly call them out in the opening post.

It's just Achievements (are there any options needed for it?) and the remaining items in Graphics -> Advanced.  Show Statistics. Show Performance Statistics. Texture Format Overlay (is that even OSD?). Unless I missed something.

iwubcode commented:
Thanks for the explanation.

> Texture Format Overlay (is that even OSD?)

That is not OSD.  It overlays on top of the textures, showing the format

mbc07 commented:
Ok, tried the most recent version of this PR. Some more thoughts:

* All controls from "Movie Window" group are dependant of the "Show Movie Window" option. To better reflect that, could you change them all to disabled state when "Show Movie Window" is unchecked?

* "Graph update rate (ms)" is only used when "Show Performance Graph" is enabled, so like the previous point, please change its state to disabled if "Show Performance Graph" is unchecked too;

* About "Log Render Time", it's not related to OSD at all, so I'd move that back to Graphics => Advanced (probably relocate it to the Debug group so it's not left as the sole option under its previous "performance statistics" group);

* Speaking of Graphics => Advanced => Debug group, we have "Show statistics" and "Show projection statistics" there, which enables specific OSD components. Those two should probably be relocated to a new "Debug" group on the On Screen Display panel this PR introduces, as they enable/disable things from the OSD;

* About moving the NetPlay group above the Movie Window group: I suggested this change because I'm under the impression that NetPlay is more used than the TAS Tools, but I don't have any statistical data to back this up. So, either order is fine...

mbc07 commented:
Another inconsistency: on this PR, if you go to Movie => Enable Movie Window, the matching option (Config => OSD => Show Movie Window) is enabled. However, if you disable the option on the OSD panel, the dropdown menu retains the old value (enabled/disabled). 

The inverse also applies, if Movie => Enable Movie Window is unchecked and you go to Config => OSD and check "Show Movie Window", the dropdown menu remains disabled...

TryTwo commented:
Thanks for catching all that! Addressed everything except the Debug group, which I'll get to.

TryTwo commented:
Added https://bugs.dolphin-emu.org/issues/13335 - movable performance metrics.  I was concerned about leaving the option enabled and accidentally clicking the UI, so I allowed the changes to persist with the option off and added a reset button.  Could also make a hotkey for moving the UI instead.

TryTwo commented:
I don't need to do it here, but as this is currently marked as a draft, I thought I'd test it out.

TryTwo commented:
I removed the options for relocating OSD stuff - like you said it needs more work.
Rebased and removed draft status. Could use a review.

mbc07 reviewed (approved):
Retested the most recent build. LGTM

sepalani reviewed (commented):
IIRC, the proper spelling of OSD uses a "-". Regarding the length of it, there are other languages with longer entries in the menu (e.g. Deutsch). I also think not using its abbreviation makes it clearer for non-technical people.

If we really have to use an abbreviation, I believe that HUD is more common than OSD. Both terms seem widely used in automotive and FPV drone even if HUD has a broader meaning (i.e. doesn't necessarily imply a screen, for instance, with cars' HUD).

Otherwise, LGTM. I really like the idea of this to have a dedicated menu, tested on Windows 11.
